Add TVLT (#20725)
huggingface/transformers: docs/source/en/model_doc/tvlt.mdx

Tips:

- TVLT is a model that takes both `pixel_values` and `audio_values` as input. One can use [`TvltProcessor`] to prepare data for the model.
  This processor wraps an image processor (for the image/video modality) and an audio feature extractor (for the audio modality) into one.
- TVLT is trained with images/videos and audios of various sizes: the authors resize and crop the input images/videos to 224 and limit the length of audio spectrogram to 2048. To make batching of videos and audios possible, the authors use a `pixel_mask` that indicates which pixels are real/padding and `audio_masks` that indicates which audio values are real/padding.

> - The design of TVLT is very similar to that of a standard Vision Transformer (ViT) and masked autoencoder (MAE). The difference is that the model includes embedding layers for the audio modality.

Reviewer: ```suggestion
- The design of TVLT is very similar to that of a standard Vision Transformer (ViT) and masked autoencoder (MAE) as in [ViTMAE](vitmae). The difference is that the model includes embedding layers for the audio modality.
```
We can link to the ViTMAE docs here

- The PyTorch version of this model is only available in torch 1.10 and higher.

<p align="center">
<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/tvlt_architecture.png"
alt="drawing" width="600"/>
</p>

<small> TVLT architecture. Taken from the <a href="[https://arxiv.org/abs/2102.03334](https://arxiv.org/abs/2209.14156)">original paper</a>. </small>

The original code can be found [here](https://github.com/zinengtang/TVLT). This model was contributed by [Zineng Tang](https://huggingface.co/TVLT).

## TvltConfig

[[autodoc]] TvltConfig

## TvltProcessor

[[autodoc]] TvltProcessor
    - __call__

## TvltImageProcessor

[[autodoc]] TvltImageProcessor

>     - __call__

Reviewer 2: ```suggestion
    - preprocess
```

The abstract from the paper is the following:

*In this work, we present the Textless Vision-Language Transformer (TVLT), where homogeneous transformer blocks take raw visual and audio inputs for vision-and-language representation learning with minimal modality-specific design, and do not use text-specific modules such as tokenization or automatic speech recognition (ASR). TVLT is trained by reconstructing masked patches of continuous video frames and audio spectrograms (masked autoencoding) and contrastive modeling to align video and audio. TVLT attains performance comparable to its text-based counterpart on various multimodal tasks, such as visual question answering, image retrieval, video retrieval, and multimodal sentiment analysis, with 28x faster inference speed and only 1/3 of the parameters. Our findings suggest the possibility of learning compact and efficient visual-linguistic representations from low-level visual and audio signals without assuming the prior existence of text.*

Tips:

- The quickest way to get started with TVLT is by checking the [example notebooks](https://github.com/zinengtang/TVLT)

>   (which showcase MAE and 2 downstream tasks in the original github).

Reviewer: Would be great to have a notebook illustrating the HF implementation of TVLT for audio/video reconstruction etc.

As it's a bit weird to refer to "getting started with TVLT" and then link to a non-HF implementation

Reviewer 3: Yes, this is already linked as the original code below. This link/tip should be removed here.

- TVLT is a model that takes both `pixel_values` and `audio_values` as input. One can use [`TvltProcessor`] to prepare data for the model.

>   This processor wraps a image processor (for the image/video modality) and a audio feature extractor (for the audio modality) into one.

Reviewer: ```suggestion
  This processor wraps an image processor (for the image/video modality) and an audio feature extractor (for the audio modality) into one.
```

> - TVLT is trained with images/videos and audios of various sizes: the authors resize and crop the input images/videos to 224 and limit the length of audio spectrogram to 2048. To make batching of videos and audios possible, the authors use a `pixel_mask` that indicates image/video masks and `audio_masks` that indicates audio makss.

Reviewer: ```suggestion
- TVLT is trained with images/videos and audios of various sizes: the authors resize and crop the input images/videos to 224 and limit the length of audio spectrogram to 2048. To make batching of videos and audios possible, the authors use a `pixel_mask` that indicates which pixels are real/padding and `audio_masks` that indicates which audio values are real/padding.
```

- The design of TVLT is very similar to that of a standard Vision Transformer (ViT) and masked autoencoder (MAE). The difference is that the model includes embedding layers for the audio modality.

<p align="center">

> <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/tvlt_architecture.jpg"

Reviewer: Looks like this URL doesn't exist yet.

Feel free to open a PR here: https://huggingface.co/datasets/huggingface/documentation-images

alt="drawing" width="600"/>
</p>

<small> TVLT architecture. Taken from the <a href="[https://arxiv.org/abs/2102.03334](https://arxiv.org/abs/2209.14156)">original paper</a>. </small>

> The original code can be found [here](https://github.com/zinengtang/TVLT).

Reviewer: ```suggestion
The original code can be found [here](https://github.com/zinengtang/TVLT). This model was contributed by [username](https://huggingface.co/username).
```
Please also add "this model was contributed by" here with your HF username.

Tips:

> - The PyTorch version of this model is only available in torch 1.10 and higher.

Reviewer: There's already a tips section above
